### DISCUSSION TITLE
Upgraded from 3.0 to 3.2 and now @svg() doesn't work

### DISCUSSION THREAD

**Discussion by Tomodo531** [statamic/cms#6330] [Troubleshooting] [answered]
I upgraded my site from 3.0 to 3.2 and now the @svg() tag on my custom login page doesn't work.

```
@inject('str', 'Statamic\Support\Str')
@extends('statamic::outside')
@section('title', __('Log in'))

@section('content')
<div class="max-w-xs mx-auto p-4">
    @svg('logo')
</div>
```
When I downgrades to 3.0 from 3.2 it works fine. Has the @svg() been removed in 3.2? I've also tried to use the SVG tags from the docs https://statamic.dev/tags/svg.
```
// Using resources/svg/circle.svg
{{ svg src="circle" class="fill-current text-teal" }}
 
// Using public/img/icons/square
{{ svg src="img/icons/square" class="fill-current text-mint" }} 
```
But they throw the following error 
`syntax error, unexpected 'src' (T_STRING), expecting ')' `

**Comment by duncanmcclean** [comment, marked as answer]
Not sure about the tag but the `@svg` Blade directive has been renamed to `@cp_svg`, after it conflicted with some popular Laravel packages.

**Reply by Tomodo531** [reply]
It somewhat helped it doesn't output @svg('logo') as text anymore but still no SVG. Do I need to specify the path somewhere I assumed it looked in `public/vendor/statamic/cp/svg/logo.svg `

**Reply by jasonvarga** [reply]
Are you using `@cp_svg` or `@svg`?

**Reply by Tomodo531** [reply]
We have found the problem the path in the svg() function has changed from:
(3.0)
```
$svg = StaticStringy::collapseWhitespace(
            File::get(public_path("vendor/statamic/cp/svg/{$name}.svg"))
        );
```
to
(3.2)
```
$svg = StaticStringy::collapseWhitespace(
            File::get(statamic_path("resources/svg/{$name}.svg"))
        );
```
So our files are the wrong place. we've hardcoded the svg for now.

**Reply by jasonvarga** [reply]
The `@svg` directive was never really intended to be used outside of the control panel. You'd be better off using your own directive or method.

**Reply by Tomodo531** [reply]
We used it to put a logo at the top of the login.blade.php page. But i quess we can try and use
```
'custom_logo_url' => [
    'nav' => '/logo-white.png',
    'outside' => '/logo-dark.png'
],
```
But what directory does it refer to or does it just output a string?

**Reply by jasonvarga** [reply]
It's expecting a URL so it'll be relative to the `public` directory.
